### PR TITLE
Adding method for dumping anthos upgrade logs

### PR DESCRIPTION
**Pull request by jainsmit**
**What this PR does / why we need it**:
Add methods to dump anthos upgrade logs

**Which issue(s) this PR fixes** (optional)
Closes # PTX-21733

**Special notes for your reviewer**:
https://jenkins.pwx.dev.purestorage.com/job/Torpedo/job/tp-anthos-upgrade/314/
```
2024-01-14 07:46:56 +0000:[DEBUG] [{UpgradeCluster}] [anthos.(*anthos).startLogCollector.func1:#733] - Collecting upgrade logs at: 2024-01-14 07:46:56.006515425 +0000 UTC m=+1962.951819625
INFO[2024-01-14 07:54:15] sched stats: workers: current=10, started=10, exited=0 
INFO[2024-01-14 07:54:15] sched stats: tick interval: 2402 < 1.2s (100%) 
INFO[2024-01-14 07:54:15] sched stats: tick process durations: 2401 < 500ms (100%) 
2024-01-14 08:01:56 +0000:[DEBUG] [{UpgradeCluster}] [anthos.(*anthos).startLogCollector.func1:#733] - Collecting upgrade logs at: 2024-01-14 08:01:56.005673319 +0000 UTC m=+2862.950977560
INFO[2024-01-14 08:04:16] sched stats: workers: current=10, started=10, exited=0 
INFO[2024-01-14 08:04:16] sched stats: tick interval: 3003 < 1.2s (100%) 
INFO[2024-01-14 08:04:16] sched stats: tick process durations: 3002 < 500ms (100%) 
INFO[2024-01-14 08:14:16] sched stats: workers: current=10, started=10, exited=0 
INFO[2024-01-14 08:14:16] sched stats: tick interval: 3603 < 1.2s (100%) 
INFO[2024-01-14 08:14:16] sched stats: tick process durations: 3602 < 500ms (100%) 
2024-01-14 08:16:56 +0000:[DEBUG] [{UpgradeCluster}] [anthos.(*anthos).startLogCollector.func1:#733] - Collecting upgrade logs at: 2024-01-14 08:16:56.00557391 +0000 UTC m=+3762.950878164
INFO[2024-01-14 08:24:16] sched stats: workers: current=10, started=10, exited=0 
INFO[2024-01-14 08:24:16] sched stats: tick interval: 4203 < 1.2s (100%) 
INFO[2024-01-14 08:24:16] sched stats: tick process durations: 4202 < 500ms (100%) 
2024-01-14 08:31:56 +0000:[DEBUG] [{UpgradeCluster}] [anthos.(*anthos).startLogCollector.func1:#733] - Collecting upgrade logs at: 2024-01-14 08:31:56.005823483 +0000 UTC m=+4662.951127655
INFO[2024-01-14 08:34:17] sched stats: workers: current=10, started=10, exited=0 
INFO[2024-01-14 08:34:17] sched stats: tick interval: 4804 < 1.2s (100%) 
INFO[2024-01-14 08:34:17] sched stats: tick process durations: 4803 < 500ms (100%) 
INFO[2024-01-14 08:44:18] sched stats: workers: current=10, started=10, exited=0 
INFO[2024-01-14 08:44:18] sched stats: tick interval: 5405 < 1.2s (100%) 
INFO[2024-01-14 08:44:18] sched stats: tick process durations: 5404 < 500ms (100%) 
2024-01-14 08:46:56 +0000:[DEBUG] [{UpgradeCluster}] [anthos.(*anthos).startLogCollector.func1:#733] - Collecting 

INFO[2024-01-14 08:54:18] sched stats: tick process durations: 6004 < 500ms (100%) 
2024-01-14 08:55:37 +0000:[DEBUG] [{UpgradeCluster}] [anthos.(*anthos).upgradeUserCluster:#475] - Successfully upgraded the user cluster%!(EXTRA string=Successfully upgraded the user cluster)
2024-01-14 08:55:37 +0000:[DEBUG] [{UpgradeCluster}] [anthos.(*anthos).stopLogCollector:#746] - Stopping log collector
2024-01-14 08:55:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).UpgradeScheduler:#275] - Anthos user cluster took: 1h23m42.524849178s time to complete the upgrade
2024-01-14 08:55:37 +0000:[INFO] [{UpgradeCluster}] [k8s.(*K8s).parseK8SNode:#879] - Parsed node [tp-anthos-upgrade-0] as Type: Worker, Zone: , Region 
2024-01-14 08:55:37 +0000:[INFO] [{UpgradeCluster}] [k8s.(*K8s).parseK8SNode:#879] - Parsed node [tp-anthos-upgrade-1] as Type: Worker, Zone: , Region 
2024-01-14 08:55:37 +0000:[INFO] [{UpgradeCluster}] [k8s.(*K8s).parseK8SNode:#879] - Parsed node [tp-anthos-upgrade-2] as Type: Worker, Zone: , Region 
2024-01-14 08:55:37 +0000:[INFO] [{UpgradeCluster}] [k8s.(*K8s).parseK8SNode:#879] - Parsed node [tp-anthos-upgrade-3] as Type: Worker, Zone: , Region 
2024-01-14 08:55:37 +0000:[INFO] [{UpgradeCluster}] [k8s.(*K8s).parseK8SNode:#879] - Parsed node [tp-anthos-upgrade-4] as Type: Worker, Zone: , Region 
2024-01-14 08:55:38 +0000:[INFO] [{UpgradeCluster}] [k8s.(*K8s).parseK8SNode:#879] - Parsed node [tp-anthos-upgrade-5] as Type: Worker, Zone: , Region 
2024-01-14 08:55:38 +0000:[INFO] [{UpgradeCluster}] [k8s.(*K8s).parseK8SNode:#879] - Parsed node [tp-anthos-upgrade-7] as Type: Worker, Zone: , Region 
2024-01-14 08:55:38 +0000:[INFO] [{UpgradeCluster}] [k8s.(*K8s).parseK8SNode:#879] - Parsed node [tp-anthos-upgrade-8] as Type: Worker, Zone: , Region 
2024-01-14 08:55:39 +0000:[INFO] [{UpgradeCluster}] [k8s.(*K8s).parseK8SNode:#879] - Parsed node [tp-anthos-upgrade-9] as Type: Worker, Zone: , Region 
2024-01-14 08:55:39 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#570] - Validating user cluster nodes upgrade time
```

Logs files are saved in workspace
```
-rwxr--r-- 1 root root  1545910 Jan 14 07:46 clusterapi-controller-manager-kube-system-2024-01-14T07:46:56Z.log
-rwxr--r-- 1 root root   183809 Jan 14 07:46 vsphere-controller-manager-kube-system-2024-01-14T07:46:56Z.log
-rwxr--r-- 1 root root      244 Jan 14 07:46 clusterapi-controller-manager-tp-anthos-upgrade-2024-01-14T07:46:56Z.log
-rwxr--r-- 1 root root      305 Jan 14 07:46 vsphere-controller-manager-tp-anthos-upgrade-2024-01-14T07:46:56Z.log
-rwxr--r-- 1 root root  1545910 Jan 14 08:01 clusterapi-controller-manager-kube-system-2024-01-14T08:01:56Z.log
-rwxr--r-- 1 root root   183809 Jan 14 08:01 vsphere-controller-manager-kube-system-2024-01-14T08:01:56Z.log
-rwxr--r-- 1 root root  1878516 Jan 14 08:01 clusterapi-controller-manager-tp-anthos-upgrade-2024-01-14T08:01:56Z.log
-rwxr--r-- 1 root root   248494 Jan 14 08:01 vsphere-controller-manager-tp-anthos-upgrade-2024-01-14T08:01:56Z.log
-rwxr--r-- 1 root root  1546614 Jan 14 08:16 clusterapi-controller-manager-kube-system-2024-01-14T08:16:56Z.log
-rwxr--r-- 1 root root   184613 Jan 14 08:16 vsphere-controller-manager-kube-system-2024-01-14T08:16:56Z.log
-rwxr--r-- 1 root root  2292358 Jan 14 08:16 clusterapi-controller-manager-tp-anthos-upgrade-2024-01-14T08:16:56Z.log
-rwxr--r-- 1 root root   316179 Jan 14 08:16 vsphere-controller-manager-tp-anthos-upgrade-2024-01-14T08:16:56Z.log
-rwxr--r-- 1 root root  1547172 Jan 14 08:31 clusterapi-controller-manager-kube-system-2024-01-14T08:31:56Z.log
-rwxr--r-- 1 root root   184613 Jan 14 08:31 vsphere-controller-manager-kube-system-2024-01-14T08:31:56Z.log
-rwxr--r-- 1 root root  3802190 Jan 14 08:31 clusterapi-controller-manager-tp-anthos-upgrade-2024-01-14T08:31:56Z.log
-rwxr--r-- 1 root root   507823 Jan 14 08:31 vsphere-controller-manager-tp-anthos-upgrade-2024-01-14T08:31:56Z.log
-rwxr--r-- 1 root root  1547172 Jan 14 08:46 clusterapi-controller-manager-kube-system-2024-01-14T08:46:56Z.log
-rwxr--r-- 1 root root   184613 Jan 14 08:46 vsphere-controller-manager-kube-system-2024-01-14T08:46:56Z.log
-rwxr--r-- 1 root root  4956692 Jan 14 08:46 clusterapi-controller-manager-tp-anthos-upgrade-2024-01-14T08:46:56Z.log
-rwxr--r-- 1 root root   676335 Jan 14 08:46 vsphere-controller-manager-tp-anthos-upgrade-2024-01-14T08:46:56Z.log
```

Now it print time taken by each node before failing 
```
registry.k8s.io/pause:3.1] 317164}] [] [] nil}}]]
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#587] - [tp-anthos-upgrade-9] node took: [7m7s] time to upgrade the node
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#587] - [tp-anthos-upgrade-2] node took: [8m53s] time to upgrade the node
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#587] - [tp-anthos-upgrade-1] node took: [8m31s] time to upgrade the node
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#587] - [tp-anthos-upgrade-0] node took: [21m44s] time to upgrade the node
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#587] - [tp-anthos-upgrade-5] node took: [8m38s] time to upgrade the node
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#587] - [tp-anthos-upgrade-4] node took: [9m14s] time to upgrade the node
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#587] - [tp-anthos-upgrade-3] node took: [15m41s] time to upgrade the node
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#587] - [tp-anthos-upgrade-8] node took: [9m44s] time to upgrade the node
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#587] - [tp-anthos-upgrade-6] node took: [8m56s] time to upgrade the node
2024-01-13 13:02:37 +0000:[ERROR] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#596] - [tp-anthos-upgrade-0] node upgrade took: [21m44s] minutes which is longer than the expected timeout value: [15m0s]
2024-01-13 13:02:37 +0000:[ERROR] [{UpgradeCluster}] [anthos.(*anthos).checkUserClusterNodesUpgradeTime:#596] - [tp-anthos-upgrade-3] node upgrade took: [15m41s] minutes which is longer than the expected timeout value: [15m0s]
2024-01-13 13:02:37 +0000:[INFO] [{UpgradeCluster}] [tests.AfterEachTest:#3045] - >>>> FAILED TEST: {UpgradeCluster} upgrade scheduler and ensure everything is running fine
```

### CODE REVIEW

File: drivers/scheduler/anthos/anthos.go [hunk 1]
```diff
@@ -643,6 +666,88 @@ func (anth *anthos) updateFileOwnership(dirPath string) error {
...
+
+// collectUpgradeLogsInNameSpace collect upgrade logs for namespace provided
+func (anth *anthos) collectUpgradeLogsInNameSpace(k8sInstance k8s.Ops, namespace string) error {
+	clusterApiLabel := make(map[string]string, 0)
+	clusterApiLabel[clusterApiKey] = clusterApiValue
+	podList, err := k8sInstance.GetPods(namespace, clusterApiLabel)
+	if err != nil {
+		return fmt.Errorf("retrieving cluster api pod is failing with error. Err: (%v)", err)
+	}
+	// Collecting pods logs from kube-system namespace
```
Comment (Rohit-PX) [inline drivers/scheduler/anthos/anthos.go:694]: Also check `len(pod.Items) > 0` and throw appropriate error. If this check is not present, then we will write empty logs as this function will return without any error for 0 pods as well.

File: drivers/scheduler/anthos/anthos.go [hunk 2]
```diff
@@ -643,6 +666,88 @@ func (anth *anthos) updateFileOwnership(dirPath string) error {
...
+			}
+		}
+	}()
+	return logTicker
+
+}
+
+// stopLogCollector stop ticker for collecting upgrade logs
+func (anth *anthos) stopLogCollector(logTicker *time.Ticker, logChan chan bool) {
+	log.Debugf("Stopping log collector")
```
Comment (Rohit-PX) [inline drivers/scheduler/anthos/anthos.go:746]:
```suggestion
	log.Debugf("Stopping log collector at %t", time.Now())
```

**Review by Rohit-PX — CHANGES_REQUESTED** [review]
LGTM otherwise.

File: drivers/scheduler/anthos/anthos.go (hunk 1 above)
Comment (jainsmit) [reply drivers/scheduler/anthos/anthos.go:698]: Incorporated !!

File: drivers/scheduler/anthos/anthos.go (hunk 2 above)
Comment (jainsmit) [reply drivers/scheduler/anthos/anthos.go]: Incorporated !!

**Review by Rohit-PX — APPROVED** [review]
LGTM.

**Review by nikolaypopov — APPROVED** [review]

**Pull request merged**